core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -146,17 +153,17 @@ public final void removePlugin(@NonNull P plugin) {
      * Retrieve a plugin by its key.
      * @param pluginKey A key that identifies a plugin implementation
      * @return The plugin object associated to pluginKey, if registered
-     * @throws IllegalStateException If there is no plugin with the given key
+     * @throws IllegalStateException If the requested plugin does not exist
      */
     @NonNull
-    public final P getPlugin(@NonNull final String pluginKey) {
+    public final P getPlugin(@NonNull final String pluginKey) throws IllegalStateException {
```

Comment:
What's the benefit to specifying the runtime exception in the method header? Also, that won't force calling methods to catch/throw it will it?

Reply:
My intention here was just to provide clear documentation on what can happen. Since `IllegalStateException` derives from `RuntimeException`, the caller still won't need to `try`-`catch` it explicitly.

_However_, I didn't honestly know the best practice on this. Should one put runtime exceptions on the method signature, or not? Apparently, the guidance is to _not_ do this. [Here's a StackOverflow post about it](https://stackoverflow.com/a/12269443/695787), which contains this quote from Oracle documentation:

> Having to add runtime exceptions in every method declaration would reduce a program's clarity.

Anyway, these are all over the code-base at this point. I'll do a separate CR to remove them all in one pass.